src/Dialogs/Cloud/WeGlideUploadDialog.cpp

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright_License {
+
+  XCSoar Glide Computer - http://www.xcsoar.org/
+  Copyright (C) 2000-2021 The XCSoar Project
+  A detailed list of copyright holders can be found in the file "AUTHORS".
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU General Public License
+  as published by the Free Software Foundation; either version 2
+  of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program; if not, write to the Free Software
+  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+}
+*/
+
+#include "WeGlideUploadDialog.hpp"
+#include "Interface.hpp"
+#include "Cloud/weglide/WeGlideSettings.hpp"
+#include "Cloud/weglide/UploadFlight.hpp"
+#include "Operation/PluggableOperationEnvironment.hpp"
+#include "net/http/Init.hpp"
+#include "json/ParserOutputStream.hxx"
+#include "Dialogs/CoDialog.hpp"
+#include "Dialogs/Error.hpp"
+#include "UIGlobals.hpp"
+#include "Language/Language.hpp"
+#include "co/InvokeTask.hxx"
+#include "Dialogs/Message.hpp"
+#include "util/Exception.hxx"
+#include "util/ConvertString.hpp"
+#include "io/StringConverter.hpp"
+#include "LogFile.hpp"
+#include "json/Lookup.hxx"
+#include "system/Path.hpp"
+#include "LocalPath.hpp"
+#include "system/FileUtil.hpp"
+
+#include <inttypes.h>
+
+namespace WeGlide {
+
+  typedef struct _Glider {
+    uint64_t id;
+    StaticString<0x40> name;
+    StaticString<4> kind;  // 'MG' - motor glider,...
+    StaticString<10> sc_class;
+    uint64_t seats;
+  } Glider;
+
+  typedef struct _UploadResponse {
+    uint64_t flight_id;
+    Pilot pilot;
+    Glider glider;
+    // StaticString<0x40> pilot_name;
+    StaticString<0x40> scoring_date;
```

Comment:
So... this field pretends to be a date, but why is its type "String"?

Reply:
This is to make the live a little bit easier: The json response from the server is only a string - and I want it only publish as string in a message dialog - why I should double convert this?

Reply:
So this data comes from the server - but why receive a scoring date from the server, when this commit should be about uploading a file?

Reply:
This json stream is the response from server after uploading igc file. If all things are ok. So the user can also know, how the data are stored on the server or - in failing case - what happened!

Reply:
And how is the "scoring_date" useful? It neither shows how the data is stored on the server, nor does it show what happens after a failure. The client already knows the scoring date.

Reply:
No, the scoring data is the flight date inside the file, That could be months / years ago - and is not directly in the file name. In this way, he will also find out under what date he can search for the flight!

Reply:
That's a contradiction. You say the scoring date is "only inside the igc file", after claiming that the client does *not* know the scoring date. Which would logically imply that the client does not have the IGC file. But it does, because this is all about the client having the IGC file and uploading it to the server.

Reply:
Ok, we can keep going in circles: In the IGC file you can find the line
HFDTEDATE:170122,
but normally no one looks in there, especially not after a perhaps stressful flight! Often enough the loggers then got the date wrong (ok, more the older models like Filser LX20 and Co.) I would just like to see after the upload on which day my uploaded file was actually classified by the server!
This is the reason for this response to the pilot!

Reply:
If you want to know the date, look at the file! ("You" = your code, not the pilot. Your code doesn't have stresful flights.) What's the point of looking at the server's response?

Reply:
Because I don't know how the server is exactly interpreting this date string...
The only thing is to look inside und show a warning that something is misinterpretable. Btw. I could alo look if the G record correct and somethin else too. But from my point of view this is a later story.